image_proc/src/rectify.cpp

```diff
@@ -0,0 +1,142 @@
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+#include <thread>
+#include <memory>
+#include <rclcpp/rclcpp.hpp>
+#include <image_transport/image_transport.h>
+#include <image_geometry/pinhole_camera_model.h>
+#include <cv_bridge/cv_bridge.h>
+#include "visibility.h"
+#include "rectify.hpp"
+namespace image_proc {
+
+RectifyNode::RectifyNode(const rclcpp::NodeOptions& options)
+: Node("RectifyNode", options)
+{
+    auto parameter_change_cb =
+    [this](std::vector<rclcpp::Parameter> parameters) -> rcl_interfaces::msg::SetParametersResult
+    {
+      auto result = rcl_interfaces::msg::SetParametersResult();
+      result.successful = true;
+      for (auto parameter : parameters) {
+        if (parameter.get_name() == "camera_namespace") {
+          camera_namespace_ = parameter.as_string();
+          RCLCPP_INFO(get_logger(), "camera_namespace: %s ", camera_namespace_.c_str());
+          break;
+        }
+      }
+      for (auto parameter : parameters){
+        if (parameter.get_name() == "image_mono") {
+          image_topic = camera_namespace_ + parameter.as_string();
+          image_rect = camera_namespace_+"/image_rect";
+          RCLCPP_INFO(get_logger(), "image_topic: %s, image_rect: %s", image_topic.c_str(), image_rect.c_str());
+          connectCb();
+          std::lock_guard<std::mutex> lock(connect_mutex_);
+          pub_rect_ = image_transport::create_publisher(this, image_rect);
+          break;
+        }
+        if (parameter.get_name() == "image_color") {
+          image_topic = camera_namespace_ + parameter.as_string();
+          image_rect = camera_namespace_ + "/image_rect_color";
+          RCLCPP_INFO(get_logger(), "image_topic: %s, image_rect: %s", image_topic.c_str(), image_rect.c_str());
+          connectCb();
+          std::lock_guard<std::mutex> lock(connect_mutex_);
+          pub_rect_ = image_transport::create_publisher(this, image_rect);
+          break;
+        }
+      }
+      return result;
+    };
+  queue_size_ = this->declare_parameter("queue_size", 5);
+  interpolation = this->declare_parameter("interpolation", 0);
+  this->set_on_parameters_set_callback(parameter_change_cb);
+}
+
+// Handles (un)subscribing when clients (un)subscribe
+void RectifyNode::connectCb( )
+{
+  std::lock_guard<std::mutex> lock(connect_mutex_);
+  if (0)
```

Comment:
Again, what's up with this?

Reply:
Hm, the only thing I can think of is some kind of future thing, as the subscription_count functionality wasn't there before (where it always defaulted to creating subscription)? 

I'll update to use `get_subscription_count`